src/components/AppSettings.vue

```diff
@@ -195,6 +207,12 @@ export default {
 		setSettingsOpen(newValue) {
 			this.settingsOpen = newValue
 			this.$emit('update:open', newValue)
+
+			if (this.settingsOpen) {
+				this.$data.initialShowHidden = Boolean(store.state.app.settings.showHidden)
+			} else if (this.$data.initialShowHidden !== store.state.app.settings.showHidden) {
+				this.$emit('reload')
+			}
```

Comment:
I'm getting fancy here and trying to avoid full in-place reloads until they're absolutely needed, instead of using `onChangeSettingsReload()`. I'm not sure it's a good fit for NC's general UX approach.